schemas/metadata-metaschema.yml

```diff
@@ -0,0 +1,72 @@
+  description: {$ref: "http://json-schema.org/draft-06/schema#/properties/description"}
+  default: {$ref: "http://json-schema.org/draft-06/schema#/properties/default"}
+  examples: {$ref: "http://json-schema.org/draft-06/schema#/properties/examples"}
+  multipleOf: {$ref: "http://json-schema.org/draft-06/schema#/properties/multipleOf"}
+  maximum: {$ref: "http://json-schema.org/draft-06/schema#/properties/maximum"}
+  exclusiveMaximum: {$ref: "http://json-schema.org/draft-06/schema#/properties/exclusiveMaximum"}
+  minimum: {$ref: "http://json-schema.org/draft-06/schema#/properties/minimum"}
+  exclusiveMinimum: {$ref: "http://json-schema.org/draft-06/schema#/properties/exclusiveMinimum"}
+  maxLength: {$ref: "http://json-schema.org/draft-06/schema#/properties/maxLength"}
+  minLength: {$ref: "http://json-schema.org/draft-06/schema#/properties/minLength"}
+  pattern: {$ref: "http://json-schema.org/draft-06/schema#/properties/pattern"}
+  additionalItems: {$ref: "http://json-schema.org/draft-06/schema#/properties/additionalItems"}
+  items: {$ref: "http://json-schema.org/draft-06/schema#/properties/items"}
+  maxItems: {$ref: "http://json-schema.org/draft-06/schema#/properties/maxItems"}
+  minItems: {$ref: "http://json-schema.org/draft-06/schema#/properties/minItems"}
+  uniqueItems: {$ref: "http://json-schema.org/draft-06/schema#/properties/uniqueItems"}
+  contains: {$ref: "http://json-schema.org/draft-06/schema#/properties/contains"}
+  maxProperties: {$ref: "http://json-schema.org/draft-06/schema#/properties/maxProperties"}
+  minProperties: {$ref: "http://json-schema.org/draft-06/schema#/properties/minProperties"}
+  required: {$ref: "http://json-schema.org/draft-06/schema#/properties/required"}
+  additionalProperties: {$ref: "http://json-schema.org/draft-06/schema#/properties/additionalProperties"}
+  definitions: {$ref: "http://json-schema.org/draft-06/schema#/properties/definitions"}
+  properties: {$ref: "http://json-schema.org/draft-06/schema#/properties/properties"}
+  patternProperties: {$ref: "http://json-schema.org/draft-06/schema#/properties/patternProperties"}
+  dependencies: {$ref: "http://json-schema.org/draft-06/schema#/properties/dependencies"}
+  propertyNames: {$ref: "http://json-schema.org/draft-06/schema#/properties/propertyNames"}
+  const: {$ref: "http://json-schema.org/draft-06/schema#/properties/const"}
+  enum: {$ref: "http://json-schema.org/draft-06/schema#/properties/enum"}
+  type: {$ref: "http://json-schema.org/draft-06/schema#/properties/type"}
+  format: {$ref: "http://json-schema.org/draft-06/schema#/properties/format"}
+  allOf: {$ref: "http://json-schema.org/draft-06/schema#/properties/allOf"}
+  anyOf: {$ref: "http://json-schema.org/draft-06/schema#/properties/anyOf"}
+  oneOf: {$ref: "http://json-schema.org/draft-06/schema#/properties/oneOf"}
+  not: {$ref: "http://json-schema.org/draft-06/schema#/properties/not"}
+
+required:
+  - metadata
+# Note that JSON-schema allows additional properties
```

Comment:
I think you should be able to avoid duplicating the json schema top level properties (lines 29-68), by structuring this schema like this:

```
allOf:
  - {$ref: "http://json-schema.org/draft-06/schema#"}
  - <metadata-metaschema requirements>
```

See https://json-schema.org/understanding-json-schema/structuring.html for an example.

Reply:
Oh, I like that, thanks!

Reply:
you're welcome! :-)